filestore/handlers.py

```diff
@@ -142,7 +140,109 @@ def __init__(self, filename, frame_per_point=1):
             frame_per_point=frame_per_point)
 
 
-class _HdfMapsHandlerBase(_HDF5HandlerBase):
+class AreaDetectorHDF5SWMRHandler(AreaDetectorHDF5Handler):
+    """
+    Handler for the 'AD_HDF5_SWMR' spec used by Area Detectors.
+
+    In this spec, the key (i.e., HDF5 dataset path) is always
+    '/entry/data/data'.
+
+    Parameters
+    ----------
+    filename : string
+        path to HDF5 file
+    frame_per_point : integer, optional
+        number of frames to return as one datum, default 1
+    """
+    specs = {'AD_HDF5_SWMR'} | HDF5DatasetSliceHandler.specs
+
+    def open(self):
+        if self._file:
+            return
+
+        self._file = h5py.File(self._filename, 'r', swmr=True)
+
+    def __call__(self, point_number):
+        if self._dataset is not None:
```

Comment:
@tacaswell I think this might be better, it just uses the class override rather than using a variable ``self._swmr``